src/derivative_operators/convolutions.jl

```diff
@@ -79,56 +123,86 @@ end
 function convolve_BC_left!(x_temp::AbstractVector{T}, _x::BoundaryPaddedVector, A::DerivativeOperator) where {T<:Real}
     stencil = A.low_boundary_coefs
     coeff   = A.coefficients
-    for i in 1 : A.boundary_point_count
-        cur_stencil = stencil[i]
+
+    _bpc = A.boundary_point_count
+
+    if cur_stencil == []
+        _bpc = 1
+    end
+
+    for i in 1 : _bpc
+        if _bpc == 1
+            cur_stencil = A.stencil_coefs
+        else
+            cur_stencil = stencil[i]
+        end
+
         cur_coeff   = typeof(coeff)   <: AbstractVector ? coeff[i] : coeff isa Number ? coeff : true
-        xtempi = cur_coeff*cur_stencil[1]*_x.l
         cur_stencil = use_winding(A) && cur_coeff < 0 ? reverse(cur_stencil) : cur_stencil
+
+        # need to account for x.l in first interior
+        xtempi = cur_coeff*cur_stencil[1]*_x.l
         @inbounds for idx in 2:A.boundary_stencil_length
             xtempi += cur_coeff * cur_stencil[idx] * _x.u[idx-1]
         end
         x_temp[i] = xtempi
     end
-    # need to account for x.l in first interior
-    mid = div(A.stencil_length,2) + 1
-    x = _x.u
-    i = 1 + A.boundary_point_count
-    xtempi = zero(T)
-    cur_stencil = eltype(A.stencil_coefs) <: AbstractVector ? A.stencil_coefs[i-A.boundary_point_count] : A.stencil_coefs
-    cur_coeff   = typeof(coeff)   <: AbstractVector ? coeff[i-A.boundary_point_count] : coeff isa Number ? coeff : true
-    cur_stencil = use_winding(A) && cur_coeff < 0 ? reverse(cur_stencil) : cur_stencil
-    xtempi = cur_coeff*cur_stencil[1]*_x.l
-    @inbounds for idx in 2:A.stencil_length
-        xtempi += cur_coeff * cur_stencil[idx] * x[(i-1) - (mid-idx) + 1]
-    end
-    x_temp[i] = xtempi
+
+    # DO WE REALLY NEED IT?
```

Comment:
This section of code here is related to my comment on line 67.

Reply:
I running the code with and without commenting this part and its not giving a different result.